MetaMask/metamask-extension
# [FLASK] Add permission cell component

## Explanation

Fixes: https://github.com/MetaMask/metamask-extension/issues/18236

This PR will add `permission-cell` component and update the pages where Snap permissions are displayed.
Designs are done according to the design specification on Figma: https://www.figma.com/file/diidnCHx9FXj5LDiA5DZZr/Snaps-features?node-id=59-19779&t=PZhPb7oTXm2Zz1aL-0

Updated pages are: Snaps settings page, Snap installation popup, Snap update popup.
This PR will also add some refactoring because most of the logic for displaying a permission is handled by newly added component `permission-cell`.

## Screenshots/Screencaps
### Before
![Screenshot 2023-03-31 at 17 20 39](https://user-images.githubusercontent.com/13301024/229162136-88351ce7-8a26-4dc0-8ee0-2013ec2a522b.png)

### After
![Screenshot 2023-04-03 at 13 05 33](https://user-images.githubusercontent.com/13301024/229492064-c7fd5eaa-9fe1-4c6c-880e-2b10d1a99eb7.png)
![Screenshot 2023-04-03 at 12 58 23](https://user-images.githubusercontent.com/13301024/229491720-62d15db2-0d15-4e86-937d-9131cbe94314.png)
![Screenshot 2023-04-03 at 12 59 43](https://user-images.githubusercontent.com/13301024/229491747-4e7efd12-1ef2-4e04-9a00-9467ef1caef2.png)
![Screenshot 2023-04-03 at 13 00 12](https://user-images.githubusercontent.com/13301024/229491769-b296b522-cfc5-490c-98dc-66ece34f550d.png)
![Screenshot 2023-04-03 at 13 01 49](https://user-images.githubusercontent.com/13301024/229491801-bc2608aa-5ff0-45c1-ab21-2d19b0ac32b6.png)

![Screenshot 2023-03-31 at 17 06 51](https://user-images.githubusercontent.com/13301024/229158581-7924ad13-a1c5-432b-ac99-8cc8d8633a93.png)


## Manual Testing Steps
1. Try to install a snap and check the UI related to the permissions.
2. Try to update a snap and check the UI related to the permissions.
3. Go to the Snap settings page and check related permission display in both, popup and expanded view.
4. Check storybook component for different UI states with different values.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
